### PR TITLE
Implement CustomRegex detector

### PR DESCRIPTION
**Pull request by mcastorina**
This detector uses webhooks for verification.

### CODE REVIEW

**Review by ahrav — APPROVED** [review]
This is gonna be huge win for the product. Nice job

File: pkg/custom_detectors/custom_detectors.go [hunk 1]
```diff
@@ -1,151 +1,216 @@
...
-
-	return nil
-}
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*customRegexWebhook)(nil)
 
-func ValidateVerifyEndpoint(endpoint string, unsafe bool) error {
-	if len(endpoint) == 0 {
-		return fmt.Errorf("no endpoint")
+func NewWebhookCustomRegex(pb *custom_detectorspb.CustomRegex) (*customRegexWebhook, error) {
```
[the remark this answers is not in the document]
Comment (mcastorina) [reply pkg/custom_detectors/custom_detectors.go:32]: It's not used anywhere *yet*. The intention is to use it in the engine once the functionality is implemented here.

I did intentionally return an unexported type from the exported function to control initialization. The idea being if a variable exists as that type, the values must have been validated.

[the remark this answers is not in the document]
Comment (mcastorina) [reply pkg/custom_detectors/custom_detectors.go:32]: I'll add some comments to make that clear for the function.

File: pkg/custom_detectors/custom_detectors.go [hunk 2]
```diff
@@ -1,151 +1,216 @@
...
+	return c.GetKeywords()
+}
 
-		for match := range matches {
-			if _, ok := regex[match]; !ok {
-				return fmt.Errorf("body %q contains an unknown variable", b)
+// productIndices produces a permutation of indices for each length. Example:
+// productIndices(3, 2) -> [[0 0] [1 0] [2 0] [0 1] [1 1] [2 1]]. It returns
+// a slice of length no larger than maxTotalMatches.
+func productIndices(lengths ...int) [][]int {
```
[the remark this answers is not in the document]
Comment (mcastorina) [reply pkg/custom_detectors/custom_detectors.go:155]: Nice, that's way better!

**Pull request merged**